software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -251,6 +251,15 @@ public void customize() {
             String saveAsRiakConf = Urls.mergePaths(getRunDir(), "riak.conf");
             DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsRiakConf).contents(riakConfTemplate));
             commands.add(sudo("mv " + saveAsRiakConf + " " + getRiakEtcDir()));
+
+            if(!osDetails.isMac()) {
+                if(!com.google.common.base.Strings.isNullOrEmpty(entity.getConfig(RiakNode.RIAK_CONFIG_FILE))) {
+                    String configTemplate = processTemplateContents(entity.getConfig(RiakNode.RIAK_CONFIG_FILE));
+                    String saveAsConfig = Urls.mergePaths(getRunDir(), "riak_extra.conf");
+                    DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsConfig).contents(configTemplate));
+                    commands.add(sudo("cat " + saveAsConfig + " >> " + getRiakEtcDir() + "/riak.conf"));
```

Comment:
But note that if `isVersion1()` then the config file is called `app.config`. I think it's better to prepare the config once and then push it to the machine, will save you from needing to execute bash commands to append.